orchagent/orch.h

```diff
@@ -81,6 +81,7 @@ class Executor : public Selectable
 
     // Decorating Selectable
     int getFd() override { return m_selectable->getFd(); }
+    int getPri() const override { return m_selectable->getPri(); }
```

Comment:
before we hade this method how everything worked?
What getPri it was used?

Reply:
1 . This bug only found in class Executor used for orch.  It worked well because most orch is not sensitive to priority, namely the process order of TABLE . 
My next pull request will show how priority influence the aclorch . aclorch works well in such condition: ACL_TABLE is processed firstly , ACL_RULE is processed secondly. 
 Class Executor is special,  it's  derived class of selectable , but it contains a Selectable pointer m_selecrtable, which is initialized with priority. But priority of its base class is always 0. Thinking below condition, Pointer Selectable * s point to an Executor object, s->getPri() will return priority derived from Select rather than priority of m_selectable.

2、 getPri was used by  Select::(std::set<Selectable *, Select::cmp> m_ready) 
when event happends, Select::select will insert  Selectable pointer into m_ready,  Select::cmp use priority to decide the process order of Selectable pointer. Selectable pointer with higher priority will be processed earlier。
```
     for (int i = 0; i < ret; ++i)
    {
        int fd = events[i].data.fd;
        Selectable* sel = m_objects[fd];
        sel->readData();
        m_ready.insert(sel);  //insert Selectable* sel to be processed
    }
......

     if (!m_ready.empty())
    {
        auto sel = *m_ready.begin(); //  sel with highest priority

        *c = sel;

        m_ready.erase(sel);
......

```

Reply:
In this case I'd suggest to fix the design of Executor, which at the same time using inheritance and composition which is not correct. Should we disable inheritance and use compassion everywhere?

Reply:
You are right, the design of Executor looks strange, but I think it's designed for two reasons:
1、 Executor is a container of pointer Selectable and the pointer is initialized by derived class pointer of Selectable, such as SubscriberStateTable, SubscriberStateTable. In this way, Executor can process event of derived class pointer of Selectable , such as readData from subscribe table
2、 Executor need to work as a selectable， so that it can be handled by Select::select()。Executor need to be selected by Select::select so that Executor::execute() can be exec  to process doTask of orch
This may be a elegant and simple design that meets the requirements metioned above. 

So I think we can do it in two steps：
Firstly, fix the bug of priority under current design
Secondly, discuss whether it is necessary to modify the design of class Executor. After all, class Executor is wildly used in orchagent, it's better to be cautious.

Reply:
@qiluo-msft please check

Reply:
retest this please，there will be no mirror since Selectable:getPri()  has been modified to be virtual in sonic-swss-common